File: src/models/entity/Pet.ts
import { Column, Entity, PrimaryGeneratedColumn } from "typeorm";

@Entity()
export class Pet {
  @PrimaryGeneratedColumn()
  id: number

  @Column({ name: 'name', type: 'varchar' })
  name: String

  @Column({ name: 'specie', type: 'varchar' })
  specie: String

  @Column({ name: 'gender', type: 'varchar' })
  gender: String

  @Column({ name: 'size', type: 'varchar' })
  size: String

  @Column({ name: 'age', type: 'varchar' })
  age: String

  @Column({ name: 'coat_length', type: 'varchar' })
  coatLength: String

  @Column({ name: 'behavior', type: 'varchar' })
  behavior: String
}

File: src/main/java/br/com/adote/um/pet/entity/Pet.java
package br.com.adote.um.pet.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "pet")
public class Pet {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String specie;

    @Column(nullable = false)
    private String gender;

    @Column(nullable = false)
    private String size;

    @Column(nullable = false)
    private String age;

    @Column(nullable = false)
    private String coatLength;

    @Column(nullable = false)
    private String behavior;
}
